src/erasure-code/jerasure/ErasureCodeJerasure.cc

```diff
@@ -239,8 +239,12 @@ int ErasureCodeJerasureReedSolomonRAID6::parse(ErasureCodeProfile &profile,
 					       ostream *ss)
 {
   int err = ErasureCodeJerasure::parse(profile, ss);
-  profile.erase("m");
-  m = 2;
+  if (m != stoi(DEFAULT_M)) {
+    *ss << "ReedSolomonRAID6: m=" << m
+        << " must be 2 for RAID6: revert to 2" << std::endl;
+    profile.erase("m");
+    err |= to_int("m", profile, &m, DEFAULT_M, ss);
```

Comment:
my argument still holds.

Reply:
Thank you, I agree with you that we should add some warnings.

Since the `DEFAULT_M` of `ErasureCodeJerasure` is 1, and `m` will be set to 1 in `ErasureCodeJerasure::parse()` if user not provide it,  I think we still need to reset `m` to 2 here.

Reply:
i believe you set it in `ErasureCodeJerasureReedSolomonRAID6` 's constructor already. and `ErasureCodeJerasureReedSolomonRAID6::parse()` is not a static method. am i missing anything?

Reply:
if user not provide `m`, we should use the default 2 without warn.
I'll try another fix.

Reply:
> i believe you set it in ErasureCodeJerasureReedSolomonRAID6 's constructor already.

yes, the std::string `DEFAULT_M` is set in ErasureCodeJerasureReedSolomonRAID6 's constructor, but int `m` is not. `m` will be assigned in function `to_int(xxx,` .

so I push a new one:
1) if user not provide `m`, then we use the default 2 without warning;
2) if user provide `m != 2`, warning and return error;
3) if user provide `m = 2`, go without warning;

Would you be alright with that?

Reply:
no. my argument still holds. see 
https://github.com/ceph/ceph/blob/a7d88133d54c373ecb60298927c2f3185a237ab5/src/erasure-code/jerasure/ErasureCodeJerasure.cc#L67

> but int `m` is not. `m` will be assigned in function `to_int(xxx,` .

i think i explained this in https://github.com/ceph/ceph/pull/29892#discussion_r318076598

Reply:
OK, I think I realized what was going on,  the below `osd_pool_default_erasure_code_profile` will be parsed ahead and `m=1` be always passed in if user doesn't provide `m`,  so the `DEFAULT_M=2` won't be applied in `ErasureCodeJerasure::parse()` at `to_int() function`. 
```
    Option("osd_pool_default_erasure_code_profile", Option::TYPE_STR, Option::LEVEL_ADVANCED)
    .set_default("plugin=jerasure technique=reed_sol_van k=2 m=1")
    .set_description(""),
```
That is what make me confused before, thank you kefu, and I think we should return -EINVAL in this case as the code shows.

I also append a commit to fix others.

Reply:
with this change, if we drop `k=2 m=1` from `osd_pool_default_erasure_code_profile`, the default values are set as expected:
```
[root@ /clove/tana/walle/repoceph/ceph/build]# ./bin/ceph osd erasure-code-profile set testecp11 technique=reed_sol_van
[root@ /clove/tana/walle/repoceph/ceph/build]# ./bin/ceph osd erasure-code-profile get testecp11
crush-device-class=
crush-failure-domain=osd
crush-root=default
jerasure-per-chunk-alignment=false
k=7
m=3
plugin=jerasure
technique=reed_sol_van
w=8
[root@ /clove/tana/walle/repoceph/ceph/build]# ./bin/ceph osd erasure-code-profile set testecp12 technique=reed_sol_r6_op
[root@ /clove/tana/walle/repoceph/ceph/build]# ./bin/ceph osd erasure-code-profile get testecp12
crush-device-class=
crush-failure-domain=osd
crush-root=default
k=7
m=2
plugin=jerasure
technique=reed_sol_r6_op
w=8
[root@ /clove/tana/walle/repoceph/ceph/build]# ./bin/ceph osd erasure-code-profile set testecp13 technique=liber8tion
[root@ /clove/tana/walle/repoceph/ceph/build]# ./bin/ceph osd erasure-code-profile get testecp13
crush-device-class=
crush-failure-domain=osd
crush-root=default
k=2
m=2
packetsize=2048
plugin=jerasure
technique=liber8tion
w=8
```
@tchaikov so, is this looks ok to you?

Reply:
@tchaikov ping

Reply:
```c++
profile.erase("m");
     err |= to_int("m", profile, &m, DEFAULT_M, ss);
```
why are you removing "m" from profile?

> I realized that it changes the behaviors, however, `-EINVAL` is returned in this case, and the `profile` will not be created if we use command `ceph osd erasure-code-profile set xxx`, so it sounds meaningless whether `m` and `w` is corrected or not.  right?

i agree with you. what i don't like is just you **changed** the way how we parse `ErasureCodeJerasureLiber8tion`'s profile without performing sanity check against it.

Reply:
> why are you removing "m" from profile? as -EINVAL is returned anyway.

I just try to keep a unified style as others, since I'm really a little confused what you want, and now i  think I got it.

> i agree with you. what i don't like is just you changed the way how we parse ErasureCodeJerasureLiber8tion's profile without performing sanity check against it.

add sanity check and warnings.

Reply:
@tchaikov what about it now, or am i missing anything?